src/plugins/vis_builder/public/application/utils/state_management/handlers/parent_aggs.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { findLast } from 'lodash';
+import { BUCKET_TYPES, IMetricAggType, search } from '../../../../../../data/public';
+import { VisBuilderServices } from '../../../../types';
+import { RootState, Store } from '../store';
+import { setAggParamValue } from '../visualization_slice';
+
+/**
+ * Parent pipeline aggs when combined with histogram aggs need `min_doc_count` to be set appropriately to avoid an error
+ * on opensearch engine https://opensearch.org/docs/2.4/opensearch/pipeline-agg/#parent-aggregations
+ */
+export const handlerParentAggs = async (
+  store: Store,
+  state: RootState,
+  services: VisBuilderServices
+) => {
+  const {
+    visualization: { activeVisualization, indexPattern = '' },
+  } = state;
+
+  const {
+    data: {
+      indexPatterns,
+      search: { aggs: aggService },
+    },
+  } = services;
+
+  if (!activeVisualization) return state;
+
+  const aggConfigs = aggService.createAggConfigs(
+    await indexPatterns.get(indexPattern),
+    activeVisualization.aggConfigParams
+  );
+
+  // Pipeline aggs should have a valid bucket agg
+  const metricAggs = aggConfigs.aggs.filter((agg) => agg.schema === 'metric');
+  const lastParentPipelineAgg = findLast(
+    metricAggs,
+    ({ type }: { type: IMetricAggType }) => type.subtype === search.aggs.parentPipelineType
+  );
+  const lastBucket = findLast(aggConfigs.aggs, (agg) => agg.type.type === 'buckets');
```

Comment:
seems like a bit of duplication with `validate_aggregations`

Reply:
😂 I can see why you would think that because i thought the same when i first saw the logic. The key difference there is that this handler is meant as a side effect to set `min_doc_count` to 0. This should only be set if the agg configs satisfy the requirements for a parent pipeline agg. The `validate_aggregations` function however is to make sure that we satisfy the pipeline agg requirements before making a request to Opensearch to fetch the data.

I could have abstracted a portion of that logic to be shared across the two but that seemed excessive and the code is more readable this way so I left it as is.

Reply:
yep, that makes sense.